Realm/RLMApp_Private.h

```diff
@@ -36,15 +36,15 @@ typedef void(^RLMAppNotificationBlock)(RLMApp *);
 /// Subscribe to notifications for this RLMApp.
 - (RLMAppSubscriptionToken *)subscribe:(RLMAppNotificationBlock)block;
 
-+ (instancetype)appWithId:(NSString *)appId
-            configuration:(nullable RLMAppConfiguration *)configuration
-            rootDirectory:(nullable NSURL *)rootDirectory;
-
-+ (instancetype)uncachedAppWithId:(NSString *)appId
-                    configuration:(RLMAppConfiguration *)configuration
-                    rootDirectory:(nullable NSURL *)rootDirectory;
++ (instancetype)appWithConfiguration:(RLMAppConfiguration *)configuration;
 
 + (void)resetAppCache;
 @end
 
+@interface RLMAppConfiguration ()
+@property (nonatomic) NSString *appId;
```

Comment:
Maybe there are some user who would like to see this exposed https://github.com/realm/realm-swift/issues/8138